controls/V-72027.rb

```diff
@@ -49,27 +43,26 @@
 -rwxr----- 1 smithj smithj 193 Mar  5 17:06 file2
 -rw-r-x--- 1 smithj smithj 231 Mar  5 17:06 file3
 
-If any files are found with a mode more permissive than \"0750\", this is a finding."
-  tag "fix": "Set the mode on files and directories in the local interactive user
-home directory with the following command:
+If any files are found with a mode more permissive than \"0750\", this is a
+finding."
+  tag "fix": "Set the mode on files and directories in the local interactive
+user home directory with the following command:
 
 Note: The example will be for the user smithj, who has a home directory of
 \"/home/smithj\" and is a member of the users group.
 
 # chmod 0750 /home/smithj/<file>"
+  tag "fix_id": "F-78379r1_fix"
 
-  # Assumption - users' home directories created in "home"
-  # Allows for mode 750 or less permissive
-  home_dirs = command('ls -d /home/*').stdout.split("\n")
-  home_dirs.each do |home|
-    home_files = command("find #{home} -xdev ! -name '.*' -type f -perm /027").stdout.split("\n")
-    home_files.each do |filename|
-      describe file(filename) do
-        it { should_not be_writable.by('group') }
-        it { should_not be_executable.by('others') }
-        it { should_not be_writable.by('others') }
-        it { should_not be_readable.by('others') }
-      end
-    end
+  IGNORE_SHELLS = NON_INTERACTIVE_SHELLS.join('|')
+
+  findings = Set[]
+  users.where{ !shell.match(IGNORE_SHELLS) && (uid >= 1000 || uid == 0)}.entries.each do |user_info|
```

Comment:
Can you confirm that `users` contains users that are not normally cached by `sssd`? This may cause extreme network/directory service load to the point that you destroy the network if it tries to resolve every possible user.

Reply:
@trevor-vaughan The `users` resource says:
```
  # The InSpec users resources looksup all local users available on a system.
  # TODO: the current version of the users resource will use eg. /etc/passwd
  # on Linux to parse all usernames. Therefore the resource may not return
  # users managed on other systems like LDAP/ActiveDirectory. Please open
  # a feature request at https://github.com/chef/inspec if you need that
  # functionality
```
So I don't think this is a problem.

Reply:
@samcornwell OK. I'll go prod a ticket for the `sssd` `LOCAL` domain at some point since it's a lovely way to stash users on your system out of sight of all these pesky scanners.